vpnglean/glean_parser_ext/templates/cpp.jinja2

```diff
@@ -38,28 +38,31 @@ struct {{ obj.name|Camelize }}{{ suffix }} {
   {% endfor %}
 
   FfiExtra ToFfiExtra() const {
-    QVector<const char*> extraKeys;
-    QVector<const char*> extraValues;
+    const char* extraKeys[{{ obj|attr(name)|length }} + 1];
+    const char* extraValues[{{ obj|attr(name)|length }} + 1];
+    int count = 0;
     {% for item, type in obj|attr(name) %}
     if (_{{item|camelize}}.canConvert<{{type|extra_type_name}}>()) {
-      extraKeys.append("{{item}}");
+      extraKeys[count] = "{{item}}";
       {% if type == "string" %}
-      extraValues.append(_{{item|camelize}}.toString().toUtf8().constData());
+      extraValues[count] = _{{item|camelize}}.toString().toUtf8().constData();
```

Comment:
I wonder when this string is deleted. What does it keep it alive? `toString` returns a temporary obj but you don't keep it alive, and probably the dtor runs. Same for the `toUtf8`.

Reply:
Is this relevant given that I do a `std::move` below?